src/parsing/compressed_reader.rs

```diff
@@ -18,7 +18,7 @@ pub enum CompressedReader {
     #[cfg(feature = "compress")]
     Deflate(DeflateDecoder<BodyReader>),
     #[cfg(feature = "compress")]
-    Gzip(Box<GzDecoder<BodyReader>>),
+    Gzip(GzDecoder<BodyReader>),
```

Comment:
Doesn't the GzDecoder from bufread also need to be boxed?

Reply:
Avoiding the additional layer of `BufReader` internally seems to make the size difference less pronounced and hence Clippy happy.

Reply:
(`flate2::read::GzDecoder<BodyReader>` is 328 bytes large whereas `flate2::bufread::GzDecoder<BodyReader>` is 296 bytes large.)